internal/eval/evalers.go

```diff
@@ -1062,14 +1169,56 @@ func newIPIsInRangeEval(lhs, rhs Evaler) *ipIsInRangeEval {
 	return &ipIsInRangeEval{lhs: lhs, rhs: rhs}
 }
 
-func (n *ipIsInRangeEval) Eval(ctx *Context) (types.Value, error) {
-	lhs, err := evalIP(n.lhs, ctx)
+func (n *ipIsInRangeEval) Eval(env *Env) (types.Value, error) {
+	lhs, err := evalIP(n.lhs, env)
 	if err != nil {
 		return zeroValue(), err
 	}
-	rhs, err := evalIP(n.rhs, ctx)
+	rhs, err := evalIP(n.rhs, env)
 	if err != nil {
 		return zeroValue(), err
 	}
 	return types.Boolean(rhs.Contains(lhs)), nil
 }
+
+// extensionEval
+
+func newExtensionEval(name types.Path, args []Evaler) Evaler {
+	// error is not part of the cedar spec, so leaving it out of the list of extensions that can be parsed, etc
+	if name == partialErrorName && len(args) == 1 {
+		return newPartialErrorEval(args[0])
+	}
```

Comment:
Did you consider making a new node type for these errors and having an eval specific for it rather than using an extension function?